files/build_templates/per_namespace/bgp.service.j2

```diff
@@ -1,16 +1,29 @@
 [Unit]
 Description=BGP container
+{% if multi_instance == 'true' %}
+Requires=database@%i.service
 Requires=updategraph.service
+After=database@%i.service
 After=updategraph.service
+{% else %}
+Requires=updategraph.service
+After=updategraph.service
+{% endif %}
 Before=ntp-config.service
 StartLimitIntervalSec=1200
 StartLimitBurst=3
 
 [Service]
 User={{ sonicadmin_user }}
+{% if multi_instance == 'true' %}
+ExecStartPre=/usr/bin/{{docker_container_name}}.sh start %i
+ExecStart=/usr/bin/{{docker_container_name}}.sh wait %i
+ExecStop=/usr/bin/{{docker_container_name}}.sh stop %i
+{% else %}
 ExecStartPre=/usr/bin/{{docker_container_name}}.sh start
 ExecStart=/usr/bin/{{docker_container_name}}.sh wait
 ExecStop=/usr/bin/{{docker_container_name}}.sh stop
+{% endif %}
```

Comment:
We could possibly simplify all these docker service file templates and eliminate duplicate lines. For example:

```suggestion
ExecStartPre=/usr/bin/{{docker_container_name}}.sh start{% if multi_instance == 'true' %} %i{% endif %}
ExecStart=/usr/bin/{{docker_container_name}}.sh wait{% if multi_instance == 'true' %} %i{% endif %}
ExecStop=/usr/bin/{{docker_container_name}}.sh stop{% if multi_instance == 'true' %} %i{% endif %}
```

Reply:
Modified as per suggestion